build/__tools__/run-tests.sh

```diff
@@ -284,5 +287,47 @@ if [ $? == 0 ]; then
     die mock contract test succeeded though it should have failed
 fi
 
+yell test the eservice database file usage
+
+# 
+say start request test. We create a new database file as part of this test
+try pdo-test-request --no-ledger   \
+    --eservice-url http://127.0.0.1:7101 \
+    --logfile $PDO_HOME/logs/client.log --loglevel info --enclaveservice-db ${ENCLAVE_DB_FILE}
+
+
+say start integer-key contract test. We update the previously created database file as part of this test by adding a new entry
+try pdo-test-contract --no-ledger --contract integer-key \
+    --eservice http://127.0.0.1:7102 \
+    --logfile $PDO_HOME/logs/client.log --loglevel info --enclaveservice-db ${ENCLAVE_DB_FILE}
+
+
+## -----------------------------------------------------------------
+yell start pdo-create and pdo-update tests
```

Comment:
this test only handles the case when the contract creator wants to rely on the db for updates. What about a "contract-joiner"? Does that also work when the first operation is a pdo-update?  I.e., provide service info only on first invocation?  (In that case one also has to consider that the contract already defines the permissible set of services)

Also other somewhat related question: in your cases you have only one eservice per contract: what is the semantic of not  specifying an eservice when multiple services are defined for a contract?  Round-robin? Random?  (arguably, the set of services is the based on the union of requirements of participants: a particular participant might only want to engage with a subset of the services -- for trust and financial reasons ... -- and insofar there is also really a difference between the contract definition and the local preference/requirements?)

Reply:
any client who has the database file can use while invoking pdo-update. Need not be the contract creator. The question as to how the client (who only updates) gets the database file is orthogonal, we do not address that. 

the contract can have more than one provisioned enclaves. all the provisioned enclaves are added to the database by the contract creator. for example, for the pdo-create test, the client will add all the encalves listed in the pcontract.toml to the database. Similarly for test-request, and test-contract, all urls specified via command line option will be added to the database by the creator (if these eservices are already running.)

For the update, the client picks a erservice at random from the pdo file, and uses the db to identify the url. 

I agree i need to add documentation specifying the semantics of the database. Will do this.

Reply:
hmm, this is really a local database as different users participate in different sets of contracts? So the " gets the database file is orthogonal, we do not address that." seems a bit strange to me?

Reply:
yes, at this point, this is a local database sharable across contracts by a single client.

To address your second point, imo, it might not be a bad idea to store this info among the "storage services". for instance, we can have a (k, v) entry per contract with k = hash(contrac_id) and v = database info for this contract. A client can populate his local database based on contracts that are of interest to the client. I guess we can take this up in a later PR.

Reply:
i completely disagree with this last observation about putting the information in a storage service. It simply moves the complexity one more level out.

I think we need to start with a document that describes the flow of creating and using the databases. For example, a simple script that takes a URL, retrieves the information from the eservice, verifies it with the ledger, and then writes it to the database makes a lot more sense to me.